test/functional/test_framework/util.py

```diff
@@ -559,6 +559,18 @@ def mine_large_block(node, utxos=None):
     node.generate(1)
 
 
+def generate_to_height(node, target_height):
+    """Generates blocks until a given target block height has been reached.
+       To prevent timeouts, only up to 200 blocks are generated per RPC call.
+       Can be used to activate certain soft-forks (e.g. CSV, CLTV)."""
+    address = node.get_deterministic_priv_key().address
+    current_height = node.getblockcount()
+    while current_height < target_height:
+        nblocks = max(200, target_height - current_height)
+        node.generatetoaddress(nblocks=nblocks, address=address)
+        current_height = node.getblockcount()
+
```

Comment:
Could write shorter and add an assert?

```suggestion
        node.generate(nblocks)
        current_height = node.getblockcount()
    assert_equal(target_height, node.getblockcount())

```

Reply:
Couldn't we just as well remove the `getblockcount` within the loop, then assert afterward? e.g. is there any harm in asserting that nothing else is going to generate blocks behind our back?

Reply:
@MarcoFalke: Thanks, done.
@laanwj: If I understood you correctly, we'd have `current_height += nblocks` in the loop and call the `getblockcount` RPC only twice -- once before the loop to initialize current_height, and another time after the loop for the assertion, right? That sounds reasonable to me, will try that out later.